lib/vector-config-macros/src/lib.rs

```diff
@@ -33,12 +34,12 @@ mod configurable_component;
 /// Additionally, callers can specify the component type, when being used directly on the top-level configuration object
 /// for a component by specifying the component type (`source`, `transform`, or `sink`) as the sole parameter:
 ///
-/// ```no_run
+/// ```ignore
```

Comment:
Yeah, as it generate code that depends on `vector_config`, which we can't import here as it would be a circular depenedency.

I tried moving the docs (and thus the doctests) to the public re-exports of the macros in `vector_config` itself, but then Rust Analyzer doesn't properly show the docs when you hover the macros.. which felt like a worse outcome than one less doctest.